package/src/components/Attachment/AudioAttachment.tsx

```diff
@@ -218,13 +255,20 @@ export const AudioAttachment = (props: AudioAttachmentProps) => {
     },
   } = useTheme();
 
-  const progressValueInSeconds = (item.duration as number) * (item.progress as number);
+  const progressValueInSeconds = useMemo(
+    () => (item.duration as number) * (item.progress as number),
+    [item.duration, item.progress],
+  );
 
-  const progressDuration = progressValueInSeconds
-    ? progressValueInSeconds / 3600 >= 1
-      ? dayjs.duration(progressValueInSeconds, 'second').format('HH:mm:ss')
-      : dayjs.duration(progressValueInSeconds, 'second').format('mm:ss')
-    : dayjs.duration(item.duration ?? 0, 'second').format('mm:ss');
+  const progressDuration = useMemo(
+    () =>
+      progressValueInSeconds
+        ? progressValueInSeconds / 3600 >= 1
+          ? dayjs.duration(progressValueInSeconds, 'second').format('HH:mm:ss')
```

Comment:
Not at all complaining about this - just curious; can it ever happen ? Wouldn't the size limit of the attachment fail many times over before we hit a video longer than an hour ?

Reply:
I haven't tried on such a long video. Technically, we don't allow uploading video/audio files of size more than 100mb. But this is just a sanity check that was added back then if the quality if poor and its a long video and backend allows uploading that video.